### DISCUSSION TITLE
Error executing queries for integration test

### DISCUSSION THREAD

**Discussion by ssona09** [Netflix/dgs-framework#1009] [Q&A] [answered]
Edited:

Hi I can't seem to be able to test my queries using WebClient.

I want to simply assert the value of a field but I can't seem to print the String value. 

If I print System.out.println(mono) it shows as MonoMapFusable.
If I print System.out.println(mono.block()) it shows the error attached.

I also tried suggestions here https://stackoverflow.com/questions/57353856/how-to-extract-string-from-monostring-in-reactor-core 

`mono.subscribe(v -> System.out.println(v));`

but it doesnt print anything

It seems like the issue is that my `client` is not executing the query. Is there any way to fix this?

I have attached the error and my code

**Comment by srinivasankavitha** [comment]
Could you please share some code sample to reproduce? It's hard to pinpoint
just looking at that error.

**Reply by ssona09** [reply]
I edited my post thanks

**Comment by bovinemagnet** [comment, marked as answer]
Took me ages to get a client going, my fault, not the fault of the library.

I ended up creating a common function to be called by my integration tests.

```java	
public static String simpleJsonCall(String query, String JSONPath) {

		// The GraphQLResponse contains data and errors.
		Mono<GraphQLResponse> graphQLResponseMono = graphQLClient.reactiveExecuteQuery(query);
		Mono<LinkedHashMap> obj = graphQLResponseMono.flatMap(r -> {
			if (r.hasErrors()) {
				log.error("[simpleJsonCall] {}", r.getErrors().toString());
				return Mono.error(new RuntimeException(r.getErrors().toString()));
			}
			if (r.extractValue(JSONPath) == null) {
				log.error("[simpleJsonCall] No results found for {}", JSONPath);
				return Mono.empty();
			} else {
				if (r.extractValue(JSONPath) instanceof LinkedHashMap) {
					return Mono.just(r.extractValue(JSONPath));
				} else {
					log.error("[simpleJsonCall] Expected LinkedHashMap but found {}",
							r.extractValue(JSONPath).getClass().getName());
					return Mono.empty();
				}
				//return Mono.just(r.extractValue(JSONPath));
			}
		});
		// Execute call to web server.
		obj.subscribe();

		// Parse the response to a JSON string.
		ObjectMapper mapper = new ObjectMapper();
		String jsonResult = "";
		obj.metrics();
		if (obj.blockOptional().isPresent()) {
			try {
				jsonResult = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(obj.block());
			} catch (Exception e) {
				log.error(e.getMessage());
			}
			log.info(jsonResult);
		}
		return jsonResult;
	}
```

Then after using the typesafe query generator I call it like this .

```java
results = RunTestLocally.simpleJsonCall(localGraphQLQueryRequest.serialize(),"address[0].regexSuburb");
```

Then I assert the results 

```java
  assertThat(results)
	.contains("regexSuburb")
	.contains( other stuff);
```

Does this help ?

**Reply by ssona09** [reply]
Super helpful, thank you!
